### PR TITLE
1167 Refactor soft deleted related querys

### PR DESCRIPTION
**Pull request by MiguelKummetz**

### CODE REVIEW

File: services/sso/src/__tests__/auth/login.test.ts [hunk 1]
```diff
@@ -51,22 +52,39 @@ describe('Testing authentication endpoint', () => {
...
+    const response = await supertest(server).post(route).send({
+      dni: testUserData.userToDelete.dni,
+      password: testUserData.userToDelete.password,
+    })
+    expect(response.status).toBe(401)
+    expect(response.body.message).toBe('Invalid Credentials')
+    await client.query('UPDATE "user" SET deleted_at = $1 WHERE dni = $2', [
+      null,
+      testUserData.userToDelete.dni,
+    ])
```
Comment (xpan1c) [inline services/sso/src/__tests__/auth/login.test.ts:88]: This should be in `afterAll`

File: services/sso/src/__tests__/users/list.test.ts [hunk 2]
```diff
@@ -68,4 +77,36 @@ describe('Testing get users name by Id endpoint', () => {
...
+    )
+    const response = await supertest(server).get(
+      `${route}?${stringData([deletedUserId])}`
+    )
+    expect(response.status).toBe(200)
+    expect(response.body).toHaveLength(0)
+    await client.query('UPDATE "user" SET deleted_at = $1 WHERE dni = $2', [
+      null,
+      testUserData.userToDelete.dni,
+    ])
```
Comment (xpan1c) [inline services/sso/src/__tests__/users/list.test.ts:94]: This should be in `afterEach`

File: services/sso/src/__tests__/users/list.test.ts [hunk 3]
```diff
@@ -68,4 +77,36 @@ describe('Testing get users name by Id endpoint', () => {
...
+    )
+    const response = await supertest(server).get(
+      `${route}?${stringData([deletedUserId, id])}`
+    )
+    expect(response.status).toBe(200)
+    expect(response.body).toHaveLength(1)
+    await client.query('UPDATE "user" SET deleted_at = $1 WHERE dni = $2', [
+      null,
+      testUserData.userToDelete.dni,
+    ])
```
Comment (xpan1c) [inline services/sso/src/__tests__/users/list.test.ts:110]: This should be in `afterEach`

File: services/sso/src/controllers/users/list.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ export const listUsers: Middleware = async (ctx: Context) => {
   const fieldsToSelect = fields.join(', ')
   const queryParams = id.map((_, index) => `$${index + 1}`).join(', ')
   const queryResult = await client.query(
-    `SELECT  ${fieldsToSelect} FROM "user" WHERE id IN (${queryParams})`,
+    `SELECT  ${fieldsToSelect} FROM "user" WHERE id IN (${queryParams}) AND deleted_at IS NULL`,
```
Comment (xpan1c) [inline services/sso/src/controllers/users/list.ts:10]: @kevinmamaqi service wiki doesn't implement soft deletes yet, should we remove this condition?

**Review by xpan1c — CHANGES_REQUESTED** [review]

**Review by xpan1c — APPROVED** [review]

**Review by julianamanzieri — DISMISSED** [review]

**Review by julianamanzieri — APPROVED** [review]

**Pull request merged**